pkg/admission/validator/lakom/shoot.go

```diff
@@ -44,16 +50,78 @@ func findExtension(extensions []core.Extension, extensionType string) (int, core
 
 func (s *shoot) validateScopeType(fldPath *field.Path, scopeType lakom.ScopeType) field.ErrorList {
 	errList := field.ErrorList{}
-
 	if !lakom.AllowedScopes.Has(scopeType) {
 		errList = append(errList, field.NotSupported(fldPath, scopeType, lakom.AllowedScopes.UnsortedList()))
 	}
 
 	return errList
 }
 
+// TODO: This check exists in the validation of the lakom config as well. It can be extracted as a util function
+func (s *shoot) validateCosignPublicKeys(fldPath *field.Path, cosignPublicKeys []config.Key) field.ErrorList {
+	errList := field.ErrorList{}
+
+	usedNames := map[string]any{}
+	for idx, k := range cosignPublicKeys {
+		if k.Name == "" {
+			errList = append(errList, field.Required(fldPath.Index(idx), "key name should no be empty"))
+		}
+
+		if _, ok := usedNames[k.Name]; ok {
+			errList = append(errList, field.Duplicate(fldPath.Index(idx), k.Name))
+		}
+		usedNames[k.Name] = nil
+
+		if keys, err := utils.GetCosignPublicKeys([]byte(k.Key)); err != nil {
+                        errList = append(errList, field.Invalid(fldPath.Index(idx), k.Key, fmt.Sprintf("key %s could not be parsed: %s", k.Name, err)))
+		} else if len(keys) != 1 {
+			errList = append(errList, field.Invalid(fldPath.Index(idx), k.Key, fmt.Sprintf("multiple keys with the name %s", k.Name)))
+		}
+	}
+
+	return errList
+}
+
+func (s *shoot) validateTrustedKeys(ctx context.Context, fldPath *field.Path, resourceName string, resources []core.NamedResourceReference, namespace string) field.ErrorList {
+    errList := field.ErrorList{}
+
+    ref := gardencorehelper.GetResourceByName(resources, resourceName)
+    if ref == nil {
+        return append(errList, field.Invalid(fldPath, resourceName, fmt.Sprintf("failed to find referenced resource with name %s", resourceName)));
```

Comment:
There is no need to put the resource name in the `detail` parameter of the `field.Invalid` function as it is already set in the `value` parameter, otherwise the error message looks like 

```
Invalid value: "lakom-ref": failed to find referenced resource with name lakom-ref
```


You can use something like instead:
 ```suggestion
        return append(errList, field.Invalid(fldPath, resourceName, "there is no resource with this name in shoot.spec.resources"));
```